AI_Model.py

```diff
@@ -0,0 +1,42 @@
+from transformers import AutoModelForCausalLM, AutoProcessor
+import numpy as np
+from PIL import Image
+
+class AI_Model:
+    def __init__(self, model_id=None, model=None, processor=None, generation_args=None):
+        if model_id is None:
+            model_id = "microsoft/Phi-3-vision-128k-instruct" 
+        self.model_id = model_id
+        if model is None:
+            model = AutoModelForCausalLM.from_pretrained(model_id,
+                                                         cache_dir="/notebooks/my_models/phi_3_vision",
+                                                         device_map="cuda",
+                                                         trust_remote_code=True,
+                                                         torch_dtype="auto",
+                                                         _attn_implementation="eager") 
+        self.model = model
+        if processor is None:
+            processor = AutoProcessor.from_pretrained(model_id, trust_remote_code=True) 
+        self.processor = processor
+        if generation_args is None:
+            generation_args = {
+                "max_new_tokens": 1024,
+                "temperature": 0.0,
+                "do_sample": False,
+            } 
+        self.generation_args = generation_args
+    
+    def get_response(self, message, image):
```

Comment:
Ova kao i svaka druga funkcija u svakom fajlu treba da sadrzi tipove promenljivih, tip outputa i komentar ispod funkcije sta ta funkcija radi. Na ovom konkretnom primeru treba da pise:
def get_response(self, message: str, image: np.array) -> str:
# A function for response generation based on prompts and images